tasks/managed/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -133,69 +132,100 @@ spec:
-        echo -n "${fbc_opt_in}" | tee "$(results.isFbcOptIn.path)"
-
-        jq '.reason // "Unset"'  <<< "${conditions}" | tee "$(results.requestReason.path)"
-        jq '.message // "Unset"' <<< "${conditions}" | tee "$(results.requestMessage.path)"
-
-        jq -r '.indexImageDigests' <<< "${results}" |  tee "$(results.indexImageDigests.path)"
+        LENGTH=$(jq -r '.components | length' "${SNAPSHOT_PATH}")
+        for((i=0; i<LENGTH; i++)); do
+          fbc_fragment=$(jq -cr --argjson i "$i" '.components[$i].containerImage' "${SNAPSHOT_PATH}")
+          echo "Processing fragment: ${fbc_fragment}:"
+
+          # The internal-request script will create the InternalRequest and wait until it finishes to get its status
+          # If it fails (Failed, Rejected or Timed out) the script will exit and display the reason.
+          echo "Creating InternalRequest to add FBC contribution to index image:"
+          internal-request --pipeline "update-fbc-catalog" \
+              -p fromIndex="$(params.fromIndex)" \
+              -p targetIndex="${target_index}" \
+              -p fbcFragment="${fbc_fragment}" \
+              -p iibServiceAccountSecret="${iib_service_account_secret}" \
+              -p publishingCredentials="${publishing_credentials}" \
+              -p buildTimeoutSeconds="${build_timeout_seconds}" \
+              -p buildTags="${build_tags}" \
+              -p addArches="${add_arches}" \
+              -p hotfix="${hotfix}" \
+              -p stagedIndex="${staged_index}" \
+              -p taskGitUrl="$(params.taskGitUrl)" \
+              -p taskGitRevision="$(params.taskGitRevision)" \
+              --service-account "${internal_request_service_account}" \
+              -l ${pipelinerun_label}="$(params.pipelineRunUid)" \
+              -t "${request_timeout_seconds}" |tee "$(workspaces.data.path)"/ir-"$(context.taskRun.uid)"-output.log
+
+          internalRequest=$(awk -F"'" '/created/ { print $2 }' \
+            "$(workspaces.data.path)"/ir-"$(context.taskRun.uid)"-output.log)
+          echo "done (${internalRequest})"
+
+          # Fetching InternalRequest status and populating results
+          results=$(kubectl get internalrequest "${internalRequest}" -o jsonpath='{.status.results}')
+          echo "${results}" > "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.json"
+
+          # Send all results to the correct results file, using yq as it allows writing inplace.
+          export results
+          yq -i '.results += [ env(results) ]' "$RESULTS_FILE"
+
+          mustSignIndexImage=$(jq -r '.genericResult | fromjson' <<< "${results}" \
+            | jq -r '.sign_index_image' |tr -d "\n")
+          mustPublishIndexImage=$(jq -r '.genericResult | fromjson' <<< "${results}" \
+            | jq -r '.publish_index_image' |tr -d "\n")
+          fbc_opt_in=$(jq -r '.genericResult | fromjson' <<< "${results}" | jq -cr '.fbc_opt_in')
+
+          # Store the results in Tekton's results files
```

Comment:
We want to use a file for all of this, right? So these should probably be removed? Because each loop of the for cycle will override these, so you will only have the results of the last component anyway. But I understand removing this now would break stuff until we modify the other tasks, so it would be acceptable to keep this with a comment that it will be removed later.

Reply:
Exactly, that was the point. E2e can't pass if I change all now. I will open a follow-up ticket to clean up the mess that will be left after this ticket is merged.

Reply:
Can't you pair this with an e2e pr?

Reply:
It would break e2e because other tasks also uses it. That's why I prefer just keep them for now then remove all later in another ticket.

Reply:
Okay, but make the new ticket high priority please